File: src/main/scala/io/koju/autopos/ApplicationWebXml.scala
package io.koju.autopos

import io.koju.autopos.config.Constants
import org.slf4j.{Logger, LoggerFactory}
import org.springframework.boot.builder.SpringApplicationBuilder
import org.springframework.boot.web.support.SpringBootServletInitializer

class ApplicationWebXml extends SpringBootServletInitializer {

  private val log: Logger = LoggerFactory.getLogger(classOf[ApplicationWebXml])

  override def configure(application: SpringApplicationBuilder): SpringApplicationBuilder =
    application.profiles(addDefaultProfile())
    .sources(classOf[Application])

  private def addDefaultProfile(): String = {
    val profile = System.getProperty("spring.profiles.active")
    if (profile != null) {
      log.info("Running with Spring profile(s) : {}", profile)
      return profile
    }
    log.warn("No Spring profile configured, running with default configuration")
    Constants.SPRING_PROFILE_DEVELOPMENT
  }

}

File: src/main/kotlin/com/kapilkoju/autopos/ApplicationWebXml.kt
package com.kapilkoju.autopos

import com.kapilkoju.autopos.config.DefaultProfileUtil
import org.springframework.boot.builder.SpringApplicationBuilder
import org.springframework.boot.web.support.SpringBootServletInitializer

class ApplicationWebXml : SpringBootServletInitializer() {

    override fun configure(application: SpringApplicationBuilder): SpringApplicationBuilder {
        DefaultProfileUtil.addDefaultProfile(application.application())
        return application.sources(Application::class.java)
    }
}